client/src/components/App.jsx

```diff
@@ -40,11 +65,78 @@ class App extends React.Component {
     )
   }
 
-  render(){
+renderFilteredData(event) {
+  event.preventDefault();
+  this.setState({
+    selectMenu: {value: event.target.value}
+  })
+};
+
+onInputSearch(event) {
+  event.preventDefault();
+  let query = event.target.value;
+  this.setState({
+    query: query
+  });
+}
+
+filterForSearchedResults(list, word) {
+  let lowerCaseWord = word.toLowerCase();
+  return list.filter(function(review){
+    let lowerCaseReview = review.review_text.toLowerCase();
+    if (lowerCaseReview.includes(lowerCaseWord)){
+      return true;
+    } else {
+      return false;
+    }
+  });
+}
+
+  filterDataNewestHelperFunction(list) {
+    list.forEach((review) => {
+      review.time = new Date(review.review_time);
+    });
+    return list.sort((a, b) => {
+      return b.time-a.time;
+    });
+  };
+
+  filterDataOldestHelperFunction(list) {
+    list.forEach((review) => {
+      review.time = new Date(review.review_time);
+    });
+    return list.sort((a, b) => {
+      return a.time - b.time;
+    });
+  }
+
+  sortHighestRatingsFirst(list) {
+    return list.sort((a, b) => {
+      return b.overall_rating - a.overall_rating;
+    });
+  };
+ 
+  sortLowestRatingsFirst(list) {
+    return list.sort((a, b) => {
+      return a.overall_rating - b.overall_rating;
+    });
+  };
+
+  render(){ 
     return (
       <div>
-       
-        {this.state.reviews.length > 0 ? this.renderReviews() : this.renderNoReviews()}
+        <div>
+          <Header reviewsCount={this.state.reviews.length} restaurantInfo = {this.state.reviews}/>
+        </div>
+        <div>
+          <div className={style.sortingText}>
+            <SelectMenu onSelectHandler={this.renderFilteredData.bind(this)} value={this.state.selectMenu.value} />
+            <SearchBar onSearch={this.onInputSearch.bind(this)}/>
```

Comment:
You might want to move the bind functions into your constructor, makes things cleaner and you'll only have to do `this.functionName` when passing this state down